core/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -133,44 +134,49 @@ protected WritePrimaryResult shardOperationOnPrimary(BulkShardRequest request, I
             // execute item request
             final Engine.Result operationResult;
             final DocWriteResponse response;
+            BulkItemRequest replicaRequest = request.items()[requestIndex];
```

Comment:
wondering whether we should make it final, so each path will have to set it. wdyt?